tests/prepare.sh

```diff
@@ -26,65 +26,123 @@ export FLORESTA_TEMP_DIR="/tmp/floresta-func-tests.${GIT_DESCRIBE}"
 mkdir -p $FLORESTA_TEMP_DIR/binaries
 
 check_installed() {
-    if ! command -v "$1" &>/dev/null; then
-        echo "You must have $1 installed to run those tests!"
-        exit 1
+    for cmd in "$@"; do
+        if ! command -v "$cmd" &>/dev/null; then
+            echo "You must have $cmd installed to run those tests!"
+            exit 1
+        fi
+    done
+}
+
+build_core() {
+    # Download and build bitcoind only
+    mkdir -p "$FLORESTA_TEMP_DIR/binaries/build"
+    cd "$FLORESTA_TEMP_DIR/binaries/build" || exit 1
+
+    echo "Downloading and Building Bitcoin Core..."
+    git clone https://github.com/bitcoin/bitcoin
+    cd bitcoin || exit 1
+
+    # If BITCOIN_REVISION is set, check it out
+    if [ -n "$BITCOIN_REVISION" ]; then
+        # Check if the revision exists as a tag only
+        if git --no-pager tag -l | grep -q "^v$BITCOIN_REVISION$"; then
+            git checkout "v$BITCOIN_REVISION"
+        else
+            echo "bitcoin 'v$BITCOIN_REVISION' is not a valid tag."
+            exit 1
+        fi
     fi
+
+    cmake -S . -B build \
+        -DBUILD_DAEMON=ON \
```

Comment:
There is need to set all of these flags because most of them you are setting their own default value. See: https://gist.github.com/hebasto/2ef97d3a726bfce08ded9df07f7dab5e